examples/offline/atari_cql.py

```diff
@@ -80,16 +82,21 @@ def test_discrete_cql(args: argparse.Namespace = get_args()) -> None:
         scale=args.scale_obs,
         frame_stack=args.frames_stack,
     )
-    args.state_shape = env.observation_space.shape or env.observation_space.n
-    args.action_shape = env.action_space.shape or env.action_space.n
+    assert isinstance(env.action_space, Discrete)
+    space_info = SpaceInfo.from_env(env)
+    args.state_shape = space_info.observation_info.obs_shape
+    args.action_shape = space_info.action_info.action_shape
+    assert isinstance(args.state_shape, list[int] | tuple[int])
+    assert len(args.state_shape) == 3
+    c, h, w = args.state_shape
     # should be N_FRAMES x H x W
     print("Observations shape:", args.state_shape)
     print("Actions shape:", args.action_shape)
     # seed
     np.random.seed(args.seed)
     torch.manual_seed(args.seed)
     # model
-    net = QRDQN(*args.state_shape, args.action_shape, args.num_quantiles, args.device)
+    net = QRDQN(c, h, w, args.action_shape, args.num_quantiles, args.device)
```

Comment:
As above. Generally, I feel like the init of such classes should be kwargs only. If you feel like it's too much, can be done in a separate PR